### PR TITLE
Add experiments option to manifest available to ZAF

### PR DESCRIPTION
**Pull request by adammw**
:v:

/cc @zendesk/vegemite

### Description
Add experiments option to manifest available to ZAF

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-504

### Risks
* low: breaks app serialization

### CODE REVIEW

**Comment by danielbreves** [comment]
Can we add a marketplace validation to ensure we only allow uploading apps with whitelisted experiment keys?

**Comment by adammw** [comment]
💇  added validation to a whitelist, that is empty for now.

File: lib/zendesk_apps_support/validations/marketplace.rb [hunk 1]
```diff
@@ -14,6 +16,13 @@ def no_symlinks(path)
           end
           nil
         end
+
+        def no_experiments(manifest)
+          non_public_experiments = manifest.enabled_experiments - WHITELISTED_EXPERIMENTS
```
Comment (sandlerr) [inline lib/zendesk_apps_support/validations/marketplace.rb]: that's user supplied input, not necessarily `non_public` but more like `invalid`?

File: lib/zendesk_apps_support/manifest.rb [hunk 2]
```diff
@@ -66,6 +67,10 @@ def parameters
       end
     end
 
+    def enabled_experiments
+      (experiments || {}).select { |k, v| v }.keys
```
Comment (sandlerr) [inline lib/zendesk_apps_support/manifest.rb]: sorry, misunderstood.

File: lib/zendesk_apps_support/validations/marketplace.rb (hunk 1 above)
Comment (adammw) [inline lib/zendesk_apps_support/validations/marketplace.rb]: will change the name

**Pull request merged**